File: src/Backend/Common/Models/Seasons/SalesPlanData.cs
using System;

namespace Company.Common.Models.Seasons
{
    public class SalesPlanData
    {
        public int Id { get; set; }

        public int DiscountPolicyId { get; set; }

        public DiscountPolicy DiscountPolicy { get; set; }

        public DateTime CreateTime { get; set; }

        public DateTime? DeleteTime { get; set; }

        public int CreatorId { get; set; }

        public int? DeleterId { get; set; }

        public float WholesaleMarginality { get; set; }

        public float NetworkMarginality { get; set; }
    }
}

File: src/Frontend/projects/core/src/lib/models/dto/seasons/sales-plan-data.ts
export class SalesPlanData {
    public id: number;
    public discountPolicyId: number;
    public createTime: Date;
    public wholesaleMarginality: number;
    public networkMarginality: number;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }
        return Object.assign(
            new SalesPlanData(),
            obj,
            {
                createTime: obj.createTime ? new Date(obj.createTime) : null,
            },
        );
    }
}
